src/blocks/sandbox-image/imports/index.mjs

```diff
@@ -0,0 +1,324 @@
+		useMedia( currentMediaId ); // @todo so this returns a different object than the media picker; its a waste of resources to get the entire media object from the media picker and then call this function, but they have different structure... optimize somehow?
+
+	// If Debug is set to true, output some helpful information to the console for block developers to utilize media object info in their block development.
+	if ( isResolvingMedia ) {
+		if ( debug ) {
+			// eslint-disable-next-line no-console
+			console.log( 'Image Media Fetch in Progress: ', isResolvingMedia );
+		}
+		return (
+			<LoadingSpinner
+				// Required Props - none
+
+				// Customizable Props
+				text="Loading..."
+
+				// @todo decide if any of these or other props should be defined or malleable
+			/>
+		);
+	}
+
+	// If Debug is set to true, output some helpful information to the console for block developers to utilize media object info in their block development.
+
+	if ( hasResolvedMedia ) {
+		if ( debug ) {
+			// eslint-disable-next-line no-console
+			console.log( 'Image Media Fetched: ', mediaObj );
+		}
+	}
+
+	if ( mediaObj ) {
+		imgObj = getImageData( mediaObj, size, 'full' );
+
+		if ( debug ) {
+			// eslint-disable-next-line no-console
+			console.log( 'Image Object: ', imgObj );
+		}
+		if ( ! imgObj ) {
+			return <div>uh oh { size } not found</div>;
+		}
+	}
+
+	// Is an image set already?
+	const hasImage = imgObj ? true : false;
+
+	/**
+	 * If there is no image set, and the user can't edit the image show placeholder.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/components/Placeholder/
+	 */
+	if ( ! hasImage && ! canEditImage ) {
+		return (
+			<Placeholder
+				// Required Props - none
+
+				// Customizable Props
+				className="bu-components-image-placeholder"
+				label={ label }
+				instructions={ instructions }
+
+				// @todo decide if any of these or other props should be defined or malleable
+			/>
+		);
+	}
+
+	/**
+	 * If there is no image set, but the user can edit the image, show Media Placeholder.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/media-placeholder/README.md
+	 */
+	if ( ! hasImage ) {
+		return (
+			<MediaPlaceholder
+				// Required Props
+				onSelect={ ( media ) => setMediaId( media.id ) }
+				// Customizable Props
+				className="bu-components-image-mediaplaceholder"
+				labels={ labels }
+				allowedTypes={ [ 'image' ] }
+				accept="image/*"
+
+				// @todo decide if any of these or other props should be defined or malleable
+			/>
+		);
+	}
+
+	/**
+	 * Build the alt text attribute.
+	 */
+	let altText = '';
+	if ( altSource === 'alt' ) {
+		altText = imgObj.alt;
+	} else if ( altSource === 'caption' ) {
+		altText = imgObj.caption;
+	} else if ( altSource === 'title' ) {
+		altText = imgObj.title;
+	} else if ( altSource === 'description' ) {
+		altText = imgObj.description;
+	} else {
+		altText = altSource;
+	}
```

Comment:
This looks like it could move to a separate file to organize this main image component better.